packages/actionmenu/src/react/index.js

```diff
@@ -216,121 +337,30 @@ const Overlay = props => (
-    } else if (evt.key === 'ArrowDown') {
-      this.navigate('down', evt)
-    } else if (evt.key === 'ArrowUp') {
-      this.navigate('up', evt)
-    } else if (evt.key === 'Tab') {
-      this.navigateTab(evt)
-    }
-  }
-  handleDividerFocus() {
-    if (this.state.activeDirection === 'down') {
-      const newIndex = this.state.activeIndex + 1
-      const itemsCount = React.Children.count(this.props.children)
-      const activeIndex = newIndex > itemsCount - 1 ? itemsCount - 1 : newIndex
-      this.setState({ activeIndex, activeDirection: 'down' })
-    } else if (this.state.activeDirection === 'up') {
-      const newIndex = this.state.activeIndex - 1
-      const activeIndex = newIndex <= 0 ? 0 : newIndex
-      this.setState({ activeIndex, activeDirection: 'up' })
-    }
-  }
-  focusItemAtIndex(i) {
-    this.setState({ activeIndex: i })
-  }
-  focusItemAtIndexWithMouse(i) {
-    this.focusItemAtIndex(i)
-    this.setState({ isKeyboarding: false })
-  }
-  navigateOut(evt) {
-    evt.stopPropagation()
-    evt.preventDefault()
-    if (typeof this.props.onClose === 'function') this.props.onClose()
-  }
-  navigate(direction, evt) {
-    evt.stopPropagation()
-    evt.preventDefault()
-    const newIndex = this.state.activeIndex + (direction === 'up' ? -1 : 1)
-    const lastIndex = React.Children.count(this.props.children) - 1
-    const activeIndex = newIndex > lastIndex ? lastIndex : newIndex
-    this.setState({
-      activeIndex,
-      activeDirection: direction,
-      isKeyboarding: true
-    })
-  }
-  navigateTab(evt) {
-    const direction = evt.shiftKey ? 'up' : 'down'
-    const { activeIndex } = this.state
-    const lastIndex = React.Children.count(this.props.children) - 1
-    const atEdge =
-      (direction === 'up' && activeIndex === 0) ||
-      (direction === 'down' && activeIndex === lastIndex)
-
-    if (atEdge) {
-      this.navigateOut(evt)
-    } else {
-      this.navigate(direction, evt)
-    }
-  }
-  render() {
-    const { ref: innerRef, ...rest } = this.props
-
-    return (
-      <Menu
-        {...rest}
-        innerRef={innerRef}
-        onKeyDown={this.handleKeyDown}
-        role="menu"
-      >
-        {React.Children.map(this.props.children, (child, i) =>
-          React.cloneElement(child, {
-            isActive: i === this.state.activeIndex,
-            shouldFocusOnMount: this.props.shouldFocusOnMount,
-            _i: i,
-            _isKeyboarding: this.state.isKeyboarding,
-            _onItemFocus: this.focusItemAtIndex,
-            _onDividerFocus: this.handleDividerFocus,
-            _onMouseOver: this.focusItemAtIndexWithMouse,
-            _origin: this.props.origin
-          })
-        )}
-      </Menu>
-    )
-  }
-}
-ActionMenu.displayName = 'ActionMenu'
-
 ActionMenu.Item = Item
 ActionMenu.Divider = Divider
 ActionMenu.Overlay = Overlay
 ActionMenu.origins = vars.origins
 ActionMenu.propTypes = {
+  children: PropTypes.oneOfType([
+    elementOfType(Item),
```

Comment:
I found it odd that I had to do this.  PropTypes puked on a single Item child when I at first just had line 356-358.